### PR TITLE
Discuss signing key expiry

### LINKED ISSUE #84: Key Management - Signing Key Expiry
Tracking https://hackmd.io/n82ZTBv3TK2Y4rujlnW3ng for conversion to a PR

### PR DESCRIPTION
**Pull request by shizhMSFT**
Resolves #84

### CODE REVIEW

File: key-revocation.md [hunk 1]
```diff
@@ -19,6 +19,35 @@ Cons:
...
+
+Long term keys are generally associated with roots, used infrequently, kept offline when not in use, and need additional protections (e.g. [HSMs](https://en.wikipedia.org/wiki/Hardware_security_module)) against compromise. Thus long term keys SHOULD not be used as signing keys.
+
+#### Short Term
+
+Since the key has short expiry that expires in days or even in minutes, the signer has to generate the key pair and get the public key certified more often by a issuing CA. This requires a robust and secure signing infrastructure that is online and can issue certificates at a higher frequency. Community developers can choose a common trusted public CA to authenticate themselves and certify their short-term ephemeral keys.
+
+* As the key expiry is shorter, timestamping MUST be used where the artifact is expected to be consumed beyond the key expiry.
+* Lowers risks of signing key compromise due to lower blast radius, when used along with timestamping.
+* Key storage can be simpler as compromised keys have limited blast radius and does not need to be protected for longer periods.
```
Comment (priteshbandi) [inline key-revocation.md:42]: Unless all artifacts signed by the key are not timestamped or If the artifact are timestamped, all Deployers are validating timestamp with configured trusted TSAs, the compromise of short term key can have high blast radius as malicious actor can backdate the signing time and select a local TSA to sign the artifact. Thus they needs to be protected all the time(until deleted).

File: key-revocation.md [hunk 2]
```diff
@@ -19,6 +19,35 @@ Cons:
...
+
+Longer expiry can make the private key exposed for a longer time, which implies larger blast radius. In other words, more artifacts are impacted when a key with longer key duration is rescinded.
+
+Long term keys are generally associated with roots, used infrequently, kept offline when not in use, and need additional protections (e.g. [HSMs](https://en.wikipedia.org/wiki/Hardware_security_module)) against compromise. Thus long term keys SHOULD not be used as signing keys.
+
+#### Short Term
+
+Since the key has short expiry that expires in days or even in minutes, the signer has to generate the key pair and get the public key certified more often by a issuing CA. This requires a robust and secure signing infrastructure that is online and can issue certificates at a higher frequency. Community developers can choose a common trusted public CA to authenticate themselves and certify their short-term ephemeral keys.
+
+* As the key expiry is shorter, timestamping MUST be used where the artifact is expected to be consumed beyond the key expiry.
```
Comment (sudo-bmitch) [inline key-revocation.md:40]: Typically key expiry is used to limit how long a signed artifact would be considered valid before requiring it to be resigned. One advantage of this is the CA signing the key's certificate can choose how long to trust that key which gives an upper limit on how long a breached key's signed artifacts may be encountered in the wild. If we allow a TSA to arbitrarily extend that limit, that removes the control the CA has on how long signed artifacts from that key are trusted.

This is trading off the fixed trust on both the key and artifact, for a short trust on the key and a potentially unlimited trust on the artifacts that short term key signs. I personally won't want to trust a TSA extended artifact unless there's some assurance the organization issuing the original signing key has some control on the max lifetime of those signed artifacts.

**Comment by SteveLasker** [comment]
@priteshbandi, do we want to close this as this seems to be covered in the signature specs?

**Comment by SteveLasker** [comment]
@shizhMSFT, @priteshbandi what is the current proposal here? Has this been captured in the signature specs?

**Comment by gillg** [comment]
You should also take a look on my notes here : https://github.com/notaryproject/notaryproject/issues/208#issuecomment-1361520221
I really think it's a important subject, and possibly consider a build artifact as a long term signed object. Obviously, util we can revoke a key by security.

**Pull request closed without merging**